### PR TITLE
worked on the pivot subsystem

### PR DESCRIPTION
**Pull request by Kevin-OConnor**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+
+public class Pivot extends SubsystemBase{
+
+    CANSparkMax pivotMotor = new CANSparkMax(0, MotorType.kBrushless);
+    AbsoluteEncoder encoder = pivotMotor.getAbsoluteEncoder(Type.kDutyCycle);
+    
+    ArmFeedforward ff; 
+
+    public Pivot() {
+        ff = new ArmFeedforward(0, 0, 0);
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: These FF values should be converted to constants

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 2]
```diff
@@ -1,5 +1,51 @@
...
+        pivotMotor.getPIDController().setFeedbackDevice(encoder);
+        //TODO: set pid values
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java:29]: Need to set motor controller PID values and current limit

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 3]
```diff
@@ -1,5 +1,51 @@
...
+import com.revrobotics.SparkPIDController.ArbFFUnits;
+
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import static frc.robot.Constants.PivotConstants.*;
+import frc.robot.MotionProfile;
+
+public class Pivot extends SubsystemBase{
+
+    CANSparkMax pivotMotor = new CANSparkMax(0, MotorType.kBrushless);
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: CAN ID should be converted to a named constant

File: src/main/java/frc/robot/subsystems/Pivot.java (hunk 2 above)
Comment (Kevin-OConnor) [reply src/main/java/frc/robot/subsystems/Pivot.java:29]: And should probably set to brake mode

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+            feed = voltageMax;
+        } else if (-voltageMax > feed) {
+            feed = -voltageMax;
+        }
+        
+
+
+        pivotMotor.getPIDController().setReference(position, ControlType.kPosition, 0, feed, ArbFFUnits.kVoltage);
+        
+    }
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: Needs methods to get the velocity and position

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 5]
```diff
@@ -1,5 +1,66 @@
...
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.SparkAbsoluteEncoder.Type;
+import com.revrobotics.SparkPIDController.ArbFFUnits;
+
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import static frc.robot.Constants.PivotConstants.*;
+import static frc.robot.Constants.ShooterConstants.kP;
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: This ShooterConstants import should be removed

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 6]
```diff
@@ -1,5 +1,66 @@
...
+    public Pivot() {
+        ff = new ArmFeedforward(kS, kG, kV);
+        pivotMotor.getPIDController().setFeedbackDevice(encoder);
+        //TODO: set pid values
+        SparkPIDController pidController = pivotMotor.getPIDController();
+                
+        pidController.setP(kP);
+        pidController.setI(kI);
+        pidController.setD(kD);
+        pivotMotor.setSecondaryCurrentLimit(currentLimit);
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: This should set the SmartCurrentLimit not the SecondaryCurrentLimit

File: src/main/java/frc/robot/subsystems/Pivot.java [hunk 7]
```diff
@@ -1,5 +1,66 @@
...
+        pivotMotor.getPIDController().setReference(position, ControlType.kPosition, 0, feed, ArbFFUnits.kVoltage);
+        
+    }
+
+    public double getVelocity() {
+        return encoder.getVelocity();
+    }
+
+    public double getCurrentPostion();
+        return encoder.getCurrentPostion();
```
Comment (Kevin-OConnor) [inline src/main/java/frc/robot/subsystems/Pivot.java]: This isn't a method of AbsoluteEncoder, needs to be corrected

Comment (me6262) [reply src/main/java/frc/robot/subsystems/Pivot.java]: also the method its inside of doesnt even have curly braces

**Pull request merged**